src/main/java/igrad/logic/commands/requirement/RequirementEditCommand.java

```diff
@@ -0,0 +1,94 @@
+
+    public static final String MESSAGE_REQUIREMENT_EDIT_SUCCESS = "Edited Requirement: %1$s";
+    public static final String MESSAGE_REQUIREMENT_NOT_EDITED = "Edited requirement must be provided with prefix "
+                                                                    + "[" + PREFIX_NAME + "] and/or "
+                                                                    + "[" + PREFIX_CREDITS + "].";
+    public static final String MESSAGE_REQUIREMENT_EMPTY_PARAMETERS = "Please provide a non-empty alphanumeric string.";
+    public static final String MESSAGE_REQUIREMENT_SAME_PARAMETERS = "Please change the title or the credits.";
+    public static final String MESSAGE_REQUIREMENT_DUPLICATE = "This requirement already exists. "
+                                                                    + "Please rename to a different title.";
+
+    private final Title originalTitle;
+
+    private final Optional<Title> newTitle;
+
+    private final Optional<Credits> newCredits;
+
+    public RequirementEditCommand(Title originalTitle,
+                                  Optional<Title> newTitle, Optional<Credits> newCredits) {
+        requireAllNonNull(originalTitle, newTitle, newCredits);
+
+        this.originalTitle = originalTitle;
+        this.newTitle = newTitle;
+        this.newCredits = newCredits;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Requirement> requirements = model.getRequirementList();
+
+        Requirement requirementToEdit = requirements.stream()
+                                            .filter(requirement -> requirement.getTitle().equals(originalTitle))
+                                            .findFirst()
+                                            .orElseThrow(() -> new CommandException(MESSAGE_REQUIREMENT_NON_EXISTENT));
+
+        Title editedTitle = newTitle.orElse(requirementToEdit.getTitle());
+        Credits editedCredits = newCredits.orElse(requirementToEdit.getCredits());
+        Requirement editedRequirement = new Requirement(editedTitle, editedCredits, requirementToEdit.getModuleList());
+
+        // if the edited title and the edited credits are the same as before
+        if (requirementToEdit.hasSameTitle(editedRequirement) && requirementToEdit.hasSameCredits(editedRequirement)) {
+            throw new CommandException(MESSAGE_REQUIREMENT_SAME_PARAMETERS);
+        }
+
+        // if the title is edited and same as before OR if the credits is edited and same as before
+        if (newTitle.isPresent() && requirementToEdit.getTitle().equals(editedTitle)
+                || newCredits.isPresent() && requirementToEdit.getCredits().equals(editedCredits)) {
+            throw new CommandException(MESSAGE_REQUIREMENT_SAME_PARAMETERS);
+        }
+
+        // if changed title is the same as an existing title

```

Comment:
i'm not sure, this looks like a duplicate to me again as the above,  if the requirement name is duplicated, it would have been sieved from the cases above